.github/workflows/dev_pr/helpers.js

```diff
@@ -69,8 +72,44 @@ async function getJiraInfo(jiraID) {
     });
 }
 
+/**
+ * Retrieves information about a GitHub issue.
+ * @param {String} issueID
+ * @returns {Object} the information about a GitHub issue.
+ */
+ async function getGitHubInfo(github, context, issueID, pullRequestNumber) {
+    try {
+        const response = await github.issues.get({
+            issue_number: issueID,
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+        })
+        return response.data
+    } catch (error) {
+        console.log(`${error.name}: ${error.code}`);
+        return false
+    }
+}
+
+/**
+ * Given the title of a PullRequest checks if it contains a GitHub issue ID
+ * @param {String} title
+ * @returns {Boolean} true if title starts with a GitHub ID or MINOR:
+ */
+ function haveGitHubIssueID(title) {
+    if (!title) {
+      return false;
+    }
+    if (title.startsWith("MINOR: ")) {
+      return true;
+    }
+    return /^(WIP:?\s*)?(GH)-\d+/.test(title);
```

Comment:
Hmm, why not consolidate the id detection functions?
You could have a single function that returns one of these possibilities:
* `{kind: "minor"}`: for a MINOR issue
* `{kind: "jira", id: "ARROW-1234"}`: for JIRA issue ARROW-1234
* `{kind: "github", id: 5678}`: for GH issue 5678
* `null`: if the title failed matching

Reply:
I've refactored everything on `detectIssue` which returns the structure suggested.